config/puma.rb

```diff
@@ -0,0 +1,14 @@
+workers Integer(ENV['WEB_CONCURRENCY'] || 2)
+threads_count = Integer(ENV['MAX_THREADS'] || 5)
+threads threads_count, threads_count
+
+preload_app!
+
+rackup      DefaultRackup
+port        ENV['PORT']     || 3000
+environment ENV['RACK_ENV'] || 'development'
+
+on_worker_boot do
+  @sidekiq_pid ||= spawn("bundle exec sidekiq -c 2")
```

Comment:
You shouldn't need the hack anymore, Heroku allows even free apps to have a worker now.